### ISSUE TITLE
Scribble thinks address is uint

### ISSUE THREAD

**Issue by NeverFearTomorrow** [Consensys/scribble#218]
Scribble file:
```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.9;

contract Dummy5 {
  /// #if_succeeds
  ///   forall(address addr in addrs)
  ///     true;
  function f(address[] calldata addrs) public {
  }
}
```
Error upon `arm`ing:
`contracts/Dummy5.sol:6:24 TypeError: The type address of the iterator variable addr is not compatible with iterator type uint of addrs.`

It appears that `addr` is actually the uint index into the array, so I guess i should be saying `forall(uint i in addrs) blah(addrs[i])`
However, I don't think this is made clear in the error message or the documentation

**Comment by blitz-1306** [comment]
@NeverFearTomorrow Hello there. I guess we have a poor description in the [docs](https://docs.scribble.codes/language/keywords/quantification#arrays-maps). But it is there:

> The second flavor of universal quantifiers talks about all **_indices_** in an array or keys in a map.

Also error states:

> The type address of the **_iterator_** variable addr is not compatible with **_iterator type_** uint of addrs.

Maybe it is not that clear to end users though, that we talk aboun indices and not the values. I must admit that it is a common misunderstanding.

**Comment by NeverFearTomorrow** [comment]
Ok, i think the docs are probably clear, maybe i just didn't read them carefully enough.
The error message doesn't seem clear to me. I think an iterator variable could be an index `i` or it could be like a foreach iterator variable where the variable is a member of the iterated sequence rather than in index into it

**Comment by blitz-1306** [comment]
I will look into tweaking an error message in near future to clarify what Scribble expects. It would take some time while I'm preoccupied with other matters. Otherwise, @cd1m0 may also address this.

**Comment by blitz-1306** [comment]
As part of #222, I changed error message to following:
```
The type {foo} of the iterator variable {bar} is not castable to the array index type uint of {baz}.
```
Should be more concise. Fix would be vailable in next Scribble release.

**Comment by blitz-1306** [comment]
This should be fixes in [`0.6.23` release](https://github.com/Consensys/scribble/releases/tag/v0.6.23). Closing.
